pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -213,6 +216,58 @@ func (nc *NodeController) RegisterNodes(nodes *api.NodeList, retryCount int, ret
 	}
 }
 
+// reconcileExternalServices updates balancers for external services, so that they will match the nodes given.
+func (nc *NodeController) reconcileExternalServices(nodes *api.NodeList) {
+	balancer, ok := nc.cloud.TCPLoadBalancer()
+	if !ok {
+		glog.Error("The cloud provider does not support external TCP load balancers.")
+		return
+	}
+
+	zones, ok := nc.cloud.Zones()
+	if !ok {
+		glog.Error("The cloud provider does not support zone enumeration.")
+		return
+	}
+	zone, err := zones.GetZone()
+	if err != nil {
+		glog.Errorf("Error while getting zone: %v", err)
+		return
+	}
+
+	hosts := []string{}
+	for _, node := range nodes.Items {
+		hosts = append(hosts, node.Name)
+	}
+
+	services, err := nc.kubeClient.Services(api.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		glog.Errorf("Error while listing services: %v", err)
+		return
+	}
+	for _, service := range services.Items {
+		if service.Spec.CreateExternalLoadBalancer {
+			nonTCPPort := false
+			for i := range service.Spec.Ports {
+				if service.Spec.Ports[i].Protocol != api.ProtocolTCP {
+					nonTCPPort = true
+					break
+				}
+			}
+			if nonTCPPort {
+				// TODO: Support UDP here.
+				glog.Errorf("External load balancers for non TCP services are not currently supported: %v.", service)
+				continue
+			}
+			name := cloudprovider.GetLoadBalancerName(nc.clusterName, service.Namespace, service.Name)
+			err := balancer.UpdateTCPLoadBalancer(name, zone.Region, hosts)
+			if err != nil {
+				glog.Errorf("External error while updating TCP load balancer: %v.", err)
```

Comment:
In the case that this fails, it looks like it won't be retried at all until the next time a node is added or removed from the cluster. This could be quite a long time in many circumstances.

In the case that this means we're not sending traffic to one of the nodes of the cluster, it isn't a huge deal due to the second layer of load balancing in the kube-proxy, but if it means that we're sending traffic to a node that's no longer part of the cluster, it may be a bigger problem.